hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/InodeMetadataRocksDBCheckpoint.java

```diff
@@ -0,0 +1,151 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hdds.utils.db;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * RocksDB checkpoint implementation that uses hardlinks to optimize disk space
+ * for inode-based metadata checkpoints.
+ *
+ * <p>During construction, reads a hardlink mapping file and creates hardlinks
+ * from checkpoint files to the checkpoint_data directory. Original files are
+ * then deleted since they're accessible via hardlinks, saving disk space while
+ * maintaining checkpoint functionality.
+ * </p>
+ */
+public class InodeMetadataRocksDBCheckpoint implements DBCheckpoint {
+
+  private final Path checkpointLocation;
+  private final long checkpointTimestamp = System.currentTimeMillis();
+  private final long latestSequenceNumber = -1;
+  private final long checkpointCreationTimeTaken = 0L;
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(InodeMetadataRocksDBCheckpoint.class);
+
+  public static final String OM_HARDLINK_FILE = "hardLinkFile";
+
+  public InodeMetadataRocksDBCheckpoint(Path checkpointLocation) throws IOException {
+    this.checkpointLocation = checkpointLocation;
+    installHardLinks();
+  }
+
+  @Override
+  public Path getCheckpointLocation() {
+    return this.checkpointLocation;
+  }
+
+  @Override
+  public long getCheckpointTimestamp() {
+    return this.checkpointTimestamp;
+  }
+
+  @Override
+  public long getLatestSequenceNumber() {
+    return this.latestSequenceNumber;
+  }
+
+  @Override
+  public long checkpointCreationTimeTaken() {
+    return this.checkpointCreationTimeTaken;
+  }
+
+  @Override
+  public void cleanupCheckpoint() throws IOException {
+    LOG.info("Cleaning up RocksDB checkpoint at {}",
+        checkpointLocation.toString());
+    FileUtils.deleteDirectory(checkpointLocation.toFile());
+  }
+
+  private void installHardLinks() throws IOException {
+    File hardLinkFile = new File(checkpointLocation.toFile(),
+        OM_HARDLINK_FILE);
+
+    if (!hardLinkFile.exists()) {
+      LOG.error("Hardlink file : {} does not exist.", hardLinkFile);
+      return;
+    }
+
+    // Track source files that need to be deleted after hardlink creation
+    List<Path> sourceFilesToDelete = new ArrayList<>();
+
+    // Read file and create hardlinks directly in checkpointLocation
+    try (Stream<String> s = Files.lines(hardLinkFile.toPath())) {
+      List<String> lines = s.collect(Collectors.toList());
+
+      // Create hardlinks directly in checkpointLocation
+      for (String l : lines) {
+        String[] parts = l.split("\t");
```

Comment:
make the "\t" a delimiter constant and use constant both here and CreateHardLinkFile function